File: src/main/clients/openrgb/effects/Static.ts
import Device from '../client/classes/Device';
import Client from '../client/client';
import { AbstractEffect } from './base/AbstractEffect';

class Static extends AbstractEffect {
  public getName(): string {
    return 'Static';
  }

  protected async applyEffect(client: Client, devices: Array<Device>): Promise<void> {
    const color = {
      red: this.color!.red * this.brightness,
      green: this.color!.green * this.brightness,
      blue: this.color!.blue * this.brightness
    };
    devices.forEach((element, i) => {
      if (!element) return;
      client.updateLeds(i, Array(element.colors.length).fill(color));
    });
    this.hasFinished = true;
  }
}

export const staticEffect = new Static();

File: lib/clients/openrgb/effects/static.py
from lib.clients.openrgb.effects.base.abstract_effect import AbstractEffect
from lib.utils.singleton import singleton


@singleton
class StaticEffect(AbstractEffect):
    """Static effect"""

    def __init__(self):
        super().__init__("Static", "#FF0000")

    def apply_effect(self):
        for dev in self.devices:
            self._set_colors(dev, [self._color] * len(dev.leds))


static_effect = StaticEffect()
